File: 知识模块/HZUIKIT/HZUIKIT/UITableView/UITableView的布局--简单内容展示/HZNewsFrame.m
//
//  HZNewsFrame.m
//  HZThreeParty
//
//  Created by 华天杰 on 2018/10/16.
//  Copyright © 2018年 HZ. All rights reserved.
//

#import "HZNewsFrame.h"
#import "HZNews.h"

#define newsCellMargin 10.0f

@implementation HZNewsFrame
- (void)setNews:(HZNews *)news{
    _news = news;
    
    // cell的高度
    CGFloat cellHeight = 0.0;
    
    // 头像
    CGFloat avatarX = newsCellMargin;
    CGFloat avatarY = newsCellMargin;
    CGFloat avatarWidth = 30.0f;
    self.avatarImageViewFrame = CGRectMake(avatarX, avatarY, avatarWidth, avatarWidth);
    // 标题
    CGFloat titleX = CGRectGetMaxX(self.avatarImageViewFrame) + newsCellMargin;
    CGFloat titleY = newsCellMargin;
    CGFloat titleWidth = 100.0f;
    CGFloat titleHeight = 30.0f;
    self.titleLabelFrame = CGRectMake(titleX, titleY, titleWidth, titleHeight);
    //圆角按钮
    CGFloat tapButtonX = [UIScreen mainScreen].bounds.size.width - 40 - 15;
    self.tapButtonFrame = CGRectMake(tapButtonX, newsCellMargin, 40, 30);
    
    // 内容
    CGFloat contentX = newsCellMargin;
    CGFloat contentY = CGRectGetMaxY(self.avatarImageViewFrame) + newsCellMargin ;
    CGFloat contentWidth = UIScreen.mainScreen.bounds.size.width - newsCellMargin * 2;
    CGFloat contentHeight = [self textHeightWithText:news.content textWidth:contentWidth];
    self.contentLabelFrame = CGRectMake(contentX, contentY, contentWidth, contentHeight);
    cellHeight = CGRectGetMaxY(self.contentLabelFrame);
    // 详细图片
    if (news.detail_image_url.length > 0) {
        CGFloat detailImageViewX = newsCellMargin;
        CGFloat detailImageViewY = CGRectGetMaxY(self.contentLabelFrame);
        CGFloat detailImageViewWidth = 60.0f;
        CGFloat detailImageViewheight = 60.0f;
        self.detailImageViewFrame = CGRectMake(detailImageViewX, detailImageViewY, detailImageViewWidth, detailImageViewheight);
        cellHeight = CGRectGetMaxY(self.detailImageViewFrame);
    }
    self.cellHeight = cellHeight + newsCellMargin ;
}

-(CGFloat)textHeightWithText:(NSString*)text textWidth:(CGFloat)textWidth{
    // 计算高度
    CGSize size = [text boundingRectWithSize:CGSizeMake(textWidth, MAXFLOAT) options:NSStringDrawingUsesLineFragmentOrigin attributes:@{NSFontAttributeName:[UIFont systemFontOfSize:20.0]} context:nil].size ;
    CGFloat heightText = size.height+5.0f;
    return heightText;
}
@end

File: 基础-语言篇/Swift/HZSwiftUIKIT/HZSwiftUIKIT/UITableView/自定义cell展示图文-cell不等高/HZNewsFrame.swift
//
//  HZNewsFrame.swift
//  HZSwiftUIKIT
//
//  Created by 华天杰 on 2019/3/13.
//  Copyright © 2019年 HZ. All rights reserved.
//

import UIKit

class HZNewsFrame: NSObject {
    public var avatarFrame:CGRect?
    public var titleFrame:CGRect?
    public var showImageFrame:CGRect?
    public var contentFrame:CGRect?
    public var confirmFrame:CGRect?
    public var cellHeight:CGFloat?
    var screenWidth = UIScreen.main.bounds.size.width
    var margin:CGFloat = 10.0
    
    public var news:HZNews?{
        willSet(new){
            // 头像
            avatarFrame = CGRect(x: margin, y: margin, width: 40, height: 40)
            // 标题
            titleFrame = CGRect(x: avatarFrame!.maxX + margin, y: margin, width: 80, height: 40)
            // 右侧确认按钮
            confirmFrame = CGRect(x: screenWidth - 60 - margin , y:margin, width: 60, height: 40)
            // 展示图片的frame
            var contentY = titleFrame!.maxY;
            if (new?.avatar_image_url) != nil {
                showImageFrame = CGRect(x: margin, y: titleFrame!.maxY + 20.0, width: screenWidth - margin*2, height: 80)
                contentY = showImageFrame!.maxY
            }else{
                showImageFrame = CGRect.zero
            }
            // 内容
            let contentHeight = UIView.textHeight(new?.content ?? "", textWidth: screenWidth - margin*2, fontSize: 15)
            contentFrame = CGRect(x: margin, y: contentY + 15.0, width: screenWidth - margin*2, height: contentHeight + 5.0)
            
            cellHeight = contentFrame!.maxY + margin;
        }
    }
    
    
}
